src/Content/Backend/Solution/Monaco.Template.Backend.Application.Infrastructure/Monaco.Template.Backend.Application.Infrastructure.csproj

```diff
@@ -6,8 +6,14 @@
 		<Nullable>enable</Nullable>
 	</PropertyGroup>
 
+	<!--#if (templateDebug)-->
+	<PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
+		<DefineConstants>$(DefineConstants);auth;commonLibraries;filesSupport</DefineConstants>
+	</PropertyGroup>
+
+	<!--#endif-->
 	<ItemGroup>
-		<!--#if (!excludeFilesSupport)-->
+		<!--#if (filesSupport)-->
 		<ProjectReference Include="..\Monaco.Template.Backend.Common.BlobStorage\Monaco.Template.Backend.Common.BlobStorage.csproj" />
 		<!--#endif-->
 		<ProjectReference Include="..\Monaco.Template.Backend.Common.Infrastructure\Monaco.Template.Backend.Common.Infrastructure.csproj" />
```

Comment:
Should this reference be wrapped in a `<!--#if (commonLibraries)-->` block?

Reply:
I don't think it's necessary to do so.
The idea of having these different flags is to be able to select what we want to generate at any given time, but not at the expense of the dependencies, but only for the rendering of the solution. And the Common dependencies are meant to always be needed for the implementation projects (such as the API, the Worker Service and the future Azure Function).
I think it this way: I'm working on a project where I know I'll have different (more than 1) microservices, so I know for sure I will need to keep all the Common projects referenced as Nuget references from each ms. So I will create a repo for a solution containing all the Common libraries and will run Monaco's backend solution template by disabling API and WorkerService projects, leaving only the Common projects.
Later, I will create another repo for Microservice1 that is meant to contain an API and a WorkerService, so I will run again Monaco's backend solution template with the same base namespace but disabling the Common projects. This will effectively generate my API and WorkerService projects already with the common dependencies linked but will only be required to convert them to Nuget references instead of the project references.
Afterwards, I can repeat the same process for Microservice2, let's say it's just a messages consumer, so it can be only the WorkerService project. All the rest will be just like explained for MS1.
Perhaps what we can do at best is to include a switch between project reference and package reference for all the common dependencies when ` commonDependencies`  flag is `false`. This way the dev should not need to manually switch that code on their own. What do you think? Do you see any other scenario I might be missing?